### ISSUE TITLE
FuncWithArgFactoryTemplate

### ISSUE THREAD

**Issue by spirit11** [structuremap/structuremap#633]
I discovered that not only Func<TO> and Func<string, TO> are injected by default, but also Func<T1, TO> if TO has constructor with at most one argument of type T1 and optionally other arguments that are defined explicitly in container.  I didn't find a description of this behavior in documentation. It is implemented by FuncWithArgFactoryTemplate and FuncWithArgInstance classes.

Maybe similar implementation for Func<T1, T2, TO>, Func<T1, T2,...Tn, TO> (to some reasonable n) should be added by creating corresponding classes FuncWith2ArgsFactoryTemplate,  FuncWith2ArgsInstance etc.?

**Comment by jeremydmiller** [comment]
@spirit11 Yeah, unless you're up for a pull request for this, it's not going to happen. See https://jeremydmiller.com/2018/01/29/sunsetting-structuremap/
